WordPressAuthenticator/Unified Auth/View Related/Sign up/SignupMagicLinkViewController.swift

```diff
@@ -0,0 +1,197 @@
+import UIKit
+
+
+/// SignupMagicLinkViewController: step two in the signup flow.
+/// This VC prompts the user to open their email app to look for the magic link we sent.
+///
+final class SignupMagicLinkViewController: LoginViewController {
+
+    // MARK: Properties
+
+    @IBOutlet private weak var tableView: UITableView!
+    private var rows = [Row]()
+    private var errorMessage: String?
+    private var shouldChangeVoiceOverFocus: Bool = false
+
+    var emailMagicLinkSource: EmailMagicLinkSource?
+    override var sourceTag: WordPressSupportSourceTag {
+        get {
+            if let emailMagicLinkSource = emailMagicLinkSource,
+                emailMagicLinkSource == .signup {
+                return .wpComSignupMagicLink
+            }
+            return .loginMagicLink
+        }
+    }
+
+    // MARK: - Actions
+    @IBAction func handleContinueButtonTapped(_ sender: NUXButton) {
+        // TODO: add the action once the button is tapped.
+    }
+
+    // MARK: - View lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        validationCheck()
+
+        navigationItem.title = WordPressAuthenticator.shared.displayStrings.signUpTitle
+        styleNavigationBar(forUnified: true)
+
+        // Store default margin, and size table for the view.
+        defaultTableViewMargin = tableViewLeadingConstraint?.constant ?? 0
+        setTableViewMargins(forWidth: view.frame.width)
+
+        localizePrimaryButton()
+        registerTableViewCells()
+        loadRows()
+    }
+
+    /// Some last-minute gatekeeping.
+    ///
+    func validationCheck() {
+        let email = loginFields.username
+        if !email.isValidEmail() {
```

Comment:
It's not clear to me why the condition is evaluated twice. If `email.isValidEmail()` returns false, the `assert` should suffice